python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py

```diff
@@ -691,6 +700,10 @@ def aten〇logical_xor〡shape(self: List[int], other: List[int]) -> List[int]:
 def aten〇logical_not〡shape(self: List[int]) -> List[int]:
     return upstream_shape_functions.unary(self)
 
+def aten〇scaled_dot_product_attention〡dtype(query_rank_dtype: Tuple[int, int], key_rank_dtype: Tuple[int, int], value_rank_dtype: Tuple[int, int], attn_mask_rank_dtype: Optional[Tuple[int, int]] = None, dropout_p: float = 0., is_causal: bool = False, scale: Optional[float] = None) -> int:
```

Comment:
Now that all the ops are using dtype functions, all new dtype functions will need testing as well. I would do something like

```python
@check_dtype_function(_check_tensors_with_the_same_dtype(num_of_tensors=3) + [
Invocation(NonZeroDTensorWithDtype(torch.float32), NonZeroDTensorWithDtype(torch.bfloat16), NonZeroDTensorWithDtype(torch.float64)), # different width
Invocation(NonZeroDTensorWithDtype(torch.float32), NonZeroDTensorWithDtype(torch.int32), NonZeroDTensorWithDtype(torch.int64)) # different type
]) 
```

Reply:
Got it, does this work?